src/libAtomVM/nifs.c

```diff
@@ -3505,6 +3557,374 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
+    }
+    if (err == EFAULT) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "efault"));
+    }
+    if (err == EFBIG) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "efbig"));
+    }
+    if (err == EINTR) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "eintr"));
+    }
+    if (err == EINVAL) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "einval"));
+    }
+    if (err == EIO) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x3", "eio"));
+    }
+    if (err == EISDIR) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "eisdir"));
+    }
+    if (err == ELOOP) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "eloop"));
+    }
+    if (err == EMFILE) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "emfile"));
+    }
+    if (err == EMLINK) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "emlink"));
+    }
+    if (err == ENAMETOOLONG) {
+        return globalcontext_make_atom(glb, ATOM_STR("\xC", "enametoolong"));
+    }
+    if (err == ENFILE) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "enfile"));
+    }
+    if (err == ENODEV) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "enodev"));
+    }
+    if (err == ENOENT) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "enoent"));
+    }
+    if (err == ENOMEM) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "enomem"));
+    }
+    if (err == ENOSPC) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "enospc"));
+    }
+    if (err == ENOTDIR) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x7", "enotdir"));
+    }
+    if (err == ENXIO) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "enxio"));
+    }
+    if (err == EPERM) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "eperm"));
+    }
+    if (err == EPIPE) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "epipe"));
+    }
+    if (err == EROFS) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "erofs"));
+    }
+    if (err == ESPIPE) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x6", "espipe"));
+    }
+    if (err == ESRCH) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "esrch"));
+    }
+    if (err == EXDEV) {
+        return globalcontext_make_atom(glb, ATOM_STR("\x5", "exdev"));
+    }
+#else
+    UNUSED(err);
+    UNUSED(glb);
+#endif
+    return UNDEFINED_ATOM;
+}
+
+#if HAVE_OPEN && HAVE_CLOSE
+struct PosixFd {
+    int fd;
+};
+
+static void posix_fd_dtor(ErlNifEnv *caller_env, void *obj)
+{
+    UNUSED(caller_env);
+
+    struct PosixFd *fd_obj = (struct PosixFd *) obj;
+    if (fd_obj->fd) {
+        close(fd_obj->fd);
+        fd_obj->fd = 0;
+    }
+}
+
+const ErlNifResourceTypeInit PosixFdResourceTypeInit = {
+    .members = 1,
+    .dtor = posix_fd_dtor,
+};
+
+#define O_EXEC_ATOM ATOM_STR("\x6", "o_exec")
```

Comment:
The convention that we have right now is that '_ATOM` suffix is reserved for those defines that can be safely assigned to vars having term type.